chain/chain/src/types.rs

```diff
@@ -407,9 +413,38 @@ mod tests {
             2,
             b1.header.inner.epoch_id.clone(),
             approvals,
-            signer.clone(),
+            &*signer,
         );
         assert!(signer.verify(b2.hash().as_ref(), &b2.header.signature));
         assert_eq!(b2.header.inner.total_weight.to_num(), 3);
     }
 }
+
+/// Verifies that chunk's proofs in the header match the body.
+pub fn validate_chunk_proofs(
+    chunk: &ShardChunk,
```

Comment:
could it be chunk-related feature?
I mean chunk.check_validity(self)

Reply:
it needs runtime_adapter, and chunks are in near_primitives which doesn't have the interface.